Adds example projects

alexhroom opened the pull request:
This PR adds example projects generated from the python-RAT examples, and also includes the generation script.

StephenNneji reviewed (approved):
Thanks for this, please remember to address the empty _controls.json_ issue. Also can you make it so new lines are stripped when loading the json (Lines 100 and 109), some editors add a new line by default and the loading kept failing even though the content was mostly the same except for the new line. 

https://github.com/RascalSoftware/RasCAL-2/blob/50dd97860090fe71c4517e22aaa91de2c98181b4/rascal2/ui/model.py#L100-L109

alexhroom commented:
@StephenNneji I've added an `.rstrip()` to the project load but couldn't reproduce the problem you were having: do you just mean an extra newline at the end of the JSON file?

also turns out that the controls saving *did* do what you wanted (saves all defaults). I just misremembered how it worked. i've updated all the controls to match what they are in the python-RAT examples (all default except for absorption which parallelises over contrasts and uses 150 resample points)

StephenNneji commented:
> @StephenNneji I've added an `.rstrip()` to the project load but couldn't reproduce the problem you were having: do you just mean an extra newline at the end of the JSON file?

I can't replicate anymore on the old commit, not sure what changed. I might have somehow introduced an error when testing but at the time I was getting this error

>  2024-11-18 13:46:07,325 - MainThread -  rascal_log - ERROR - Failed to load project at path /home/stephen/Downloads/RasCAL-2-add-examples/examples/domains_standard_layers:
>  The controls.json file for this project is not valid.
> It may contain invalid parameter values or be invalid JSON.
>
